### PR TITLE
feat: add split function 

### LINKED ISSUE #26: Add `split` function
# Suggestion

## ⭐ Suggestion

Splits string by separator. ~(not support unicode)~

## 💻 Use Cases

```ts
const iter = split(',', '1,2,3,4,5,6');
iter.next(); // 1
iter.next(); // 2
iter.next(); // 3
iter.next(); // 4
iter.next(); // 5
iter.next(); // 6
iter.next(); // undefined


```

### PR DESCRIPTION
**Pull request by ppeeou**
Fixes #26

### CODE REVIEW

File: test/Lazy/split.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { split } from "../../src";
+
+describe("split", function () {
+  describe("sync", function () {
+    it("should return an empty array", function () {
+      const iter = split("", "");
+      expect([...iter]).toEqual([]);
+    });
+
+    it("should be splited by empty string", function () {
```
Comment (hg-pyun) [inline test/Lazy/split.spec.ts:21]: Is the word "empty" right?

File: src/Lazy/split.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  sep: string,
+  iterable: Iterable<string>,
+): IterableIterator<string>;
+
+function split(sep: string, str: Iterable<string>) {
+  if (isIterable(str)) {
+    return sync(sep, str);
+  }
+
+  throw new TypeError("iterable must be type of Iterable or AsyncIterable");
```
Comment (hg-pyun) [inline src/Lazy/split.ts:105]: Let's start with capital letters.
```suggestion
  throw new TypeError("Iterable must be type of Iterable or AsyncIterable");
```

File: src/Lazy/split.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    } else {
+      acc.push(chr);
+    }
+  }
+
+  if (acc.length > 0) {
+    yield acc.join("");
+  } else if (sep !== "") {
+    yield "";
+  }
```
[the remark this answers is not in the document]
Comment (ppeeou) [reply src/Lazy/split.ts:23]: We decided to use `+=` instead of `push`+`join` with @shine1594

File: src/Lazy/split.ts (hunk 2 above)
Comment (ppeeou) [reply src/Lazy/split.ts:105]: Thank you for your comment:)

I was thinking of the variable iterable of the current context, but I think I gave the confusion to [Iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol) 

Should I usually express the variable expression like `iterable`?

File: test/Lazy/split.spec.ts (hunk 1 above)
Comment (ppeeou) [reply test/Lazy/split.spec.ts:21]: `""` <- This value entered the separator value so I wrote this expression. 
If you have a better expression, please share it :)

**Review by shine1594 — APPROVED** [review]

File: test/Lazy/split.spec.ts (hunk 1 above)
Comment (hg-pyun) [reply test/Lazy/split.spec.ts:21]: Understand. No problem.

**Review by hg-pyun — APPROVED** [review]
LGTM

**Pull request merged**